src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -1,31 +1,30 @@
 import { connect } from 'react-redux';
 
+import { isLoggedIn } from 'platform/user/selectors';
 import { checkKeepAlive } from 'platform/user/authentication/actions';
 import {
   ssoe,
   ssoeInbound,
   hasCheckedKeepAlive,
 } from 'platform/user/authentication/selectors';
-import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
-import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
-import { ssoKeepAliveSession } from 'platform/utilities/sso';
-
-export async function checkStatus(toggleKeepAlive) {
-  await ssoKeepAliveSession();
-  if (hasSession() && !hasSessionSSO()) {
-    autoLogout();
-  } else if (!hasSession() && hasSessionSSO()) {
-    autoLogin();
-  }
+import { setForceAuth, removeForceAuth } from 'platform/utilities/sso';
 
-  toggleKeepAlive();
-}
+import { parseqs, checkAutoSession } from '../helpers';
 
 function AutoSSO(props) {
-  const { useSSOe, useInboundSSOe, hasCalledKeepAlive } = props;
+  const { useSSOe, useInboundSSOe, hasCalledKeepAlive, userLoggedIn } = props;
+  const auth = parseqs(window.location.search).auth;
+
+  if (auth && auth !== 'success') {
+    setForceAuth();
+  }
+
+  if (userLoggedIn) {
```

Comment:
Should this an `else if`, or is not possible for both conditions to be met?

Reply:
yeah thats true, they can't be logged in AND require a force authentication